### PR TITLE
Doc for customizing tracing sdk

### PR DESCRIPTION
**Pull request by cijothomas**
To get an initial feedback if this/this direction is useful:
The goal is to add customizing the sdk section to docs. The individual readme.md files for each component (esp. SDK) can be modified to be a reference doc and the contents under /docs can be the step by step tutorial.

### CODE REVIEW

File: docs/trace/customizing-the-sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+Contents from https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry#tracing-configuration
+
+
+## Context Propagation
+
+Contents from https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry#propagators
+Links to built-in instrumentations doing Context Propagation.
+
+
```
Comment (reyang) [inline docs/trace/customizing-the-sdk/README.md:15]: nit, remove these extra empty lines.

File: docs/trace/customizing-the-sdk/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Customizing OpenTelemetry .NET SDK
+
+Talks about customizing the SDK.
```
Comment (reyang) [inline docs/trace/customizing-the-sdk/README.md:3]: This might be redundant since the title is already self-explanatory.

**Review by reyang — APPROVED** [review]
I think this is a good movement.

Consider adding a link in this section https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started/README.md#learn-more when the doc is ready for consumption.

**Comment by cijothomas** [comment]
Starting point. Will iteratively add content.

**Pull request merged**